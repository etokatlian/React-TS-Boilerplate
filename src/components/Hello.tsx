import * as React from "react";

export interface HelloProps {}

export interface HelloState {}

//'HelloProps' describes the shape of props.
//'HelloState' describes the shape of props.
export class Hello extends React.Component<HelloProps, HelloState> {
  render(): Array<JSX.Element> {
    return [
      <h1>React/TS/Webpack Boilerplate</h1>,
      <h3>
        We have successfully setup our React, Webpack, and Typescript project!
      </h3>
    ];
  }
}
