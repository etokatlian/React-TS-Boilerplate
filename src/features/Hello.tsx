import * as React from 'react';

export interface HelloProps {}

export interface HelloState {}

export class Hello extends React.Component<HelloProps, HelloState> {
  public render() {
    return <h1>React/TS Boilerplate</h1>;
  }
}
