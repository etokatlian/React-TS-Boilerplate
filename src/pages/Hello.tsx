import * as React from 'react';

export interface IHelloProps {
  some?: any;
}

export interface IHelloState {
  some?: any;
}

export class Hello extends React.Component<IHelloProps, IHelloState> {
  public render() {
    return <h1>React/TS Boilerplate</h1>;
  }
}
