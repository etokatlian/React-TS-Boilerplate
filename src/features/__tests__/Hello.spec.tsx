import { shallow } from 'enzyme';
import * as React from 'react';
import { Hello } from '../Hello';

test('Hello component renders', () => {
  const HelloComponent = shallow(<Hello />);

  // Snapshot demo
  expect(HelloComponent).toMatchSnapshot();
});
