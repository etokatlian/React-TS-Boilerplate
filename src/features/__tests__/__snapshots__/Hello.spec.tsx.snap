// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello component renders 1`] = `
<h1>
  React/TS Boilerplate
</h1>
`;
